src/foam/mlang/expr/Ref.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.mlang.expr',
+  name: 'Ref',
+  extends: 'foam.mlang.AbstractExpr',
+  implements: [ 'foam.core.Serializable' ],
+
+  documentation: 'A Binary Predicate which returns reference property object',
+
+  javaImports: [
+    'foam.core.AbstractFObjectPropertyInfo',
+    'foam.core.FObject',
+    'foam.core.PropertyInfo',
+    'foam.nanos.logger.Logger',
+    'foam.nanos.logger.StdoutLogger',
+    'foam.util.StringUtil'
+  ],
+
+  properties: [
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'arg1'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      code: async function(o) {
+        return o[arg1.name + '$find'];
+      },
+      javaCode: `
+        PropertyInfo p1 = (PropertyInfo) getArg1();
+        FObject refObj = null;
+        try {
+          refObj = (FObject)obj.getClass().getMethod("find" + StringUtil.capitalize(p1.getName()), foam.core.X.class).invoke(obj, ((FObject)obj).getX());
```

Comment:
can you break this line into two? one for getMethod and another to invoke.

Reply:
fixed.